packages/cli/commands/wcp/hooks.js

```diff
@@ -21,100 +21,113 @@ const { getUser, getProjectEnvironment, updateUserLastActiveOn } = require("./ut
-            const [orgId, projectId] = wcpProjectId.split("/");
-
-            const apiKey = process.env.WCP_PROJECT_ENVIRONMENT_API_KEY;
-
-            let projectEnvironment;
-            if (apiKey) {
-                projectEnvironment = await getProjectEnvironment({ apiKey });
-            } else {
-                const isValidId = orgId && projectId;
-                if (!isValidId) {
-                    throw new Error(
-                        `It seems the project ID, specified in "webiny.project.ts" file, is invalid.`
-                    );
-                }
-
-                // If there is no API key, that means we need to retrieve the currently logged-in user.
-                const user = await getUser();
-                const project = user.projects.find(item => item.id === projectId);
-                if (!project) {
-                    throw new Error(
-                        `It seems you don't belong to the current project or the current project has been deleted.`
-                    );
-                }
-
-                projectEnvironment = await getProjectEnvironment({
-                    orgId,
-                    projectId,
-                    userId: user.id,
-                    environmentId: args.env
-                });
-            }
-
-            if (projectEnvironment.org.id !== orgId) {
-                throw new Error(
-                    `Cannot proceed with the deployment because the "${projectEnvironment.name}" project environment doesn't belong to the "${orgId}" organization. Please check your WCP project ID (currently set to "${wcpProjectId}").`
-                );
-            }
-
-            if (projectEnvironment.project.id !== projectId) {
-                throw new Error(
-                    `Cannot proceed with the deployment because the "${projectEnvironment.name}" project environment doesn't belong to the "${wcpProjectId}" project. Please check your WCP project ID (currently set to "${wcpProjectId}").`
-                );
-            }
-
-            if (projectEnvironment && projectEnvironment.status !== "enabled") {
-                throw new Error(
-                    `Cannot proceed with the deployment because the "${projectEnvironment.name}" project environment has been disabled.`
-                );
-            }
-
-            // Assign `WCP_PROJECT_ENVIRONMENT` and `WCP_PROJECT_ENVIRONMENT_API_KEY`
-            const wcpProjectEnvironment = {
-                id: projectEnvironment.id,
-                apiKey: projectEnvironment.apiKey,
-                org: { id: projectEnvironment.org.id },
-                project: { id: projectEnvironment.project.id }
-            };
-
-            process.env.WCP_PROJECT_ENVIRONMENT = encrypt(wcpProjectEnvironment);
-            process.env.WCP_PROJECT_ENVIRONMENT_API_KEY = projectEnvironment.apiKey;
-        }
+        hook: getEnvironmentHookHandler
```

Comment:
It's not a `getter` if you're not executing it :) Just rename it to `environmentHookHandler`, then it can work without being called.

Reply:
Ah jsut realized, `get` is not a "factory" for a hook, it's the name of the hook logic. The hook is _getting_ the environment. My bad.